lib/ansible/modules/web_infrastructure/ansible_tower/tower_workflow_launch.py

```diff
@@ -0,0 +1,166 @@
+options:
+    workflow_template:
+      description:
+        - The name of the workflow template to run.
+      required: True
+    extra_vars:
+      description:
+        - Any extra vars required to launch the job.
+      required: False
+    wait:
+      description:
+        - Wait for the workflow to complete.
+      required: False
+      default: True
+      type: bool
+    timeout:
+      description:
+        - If waiting for the workflow to complete this will abort after this
+        - ammount of seconds
+
+requirements:
+  - "python >= 2.6"
+extends_documentation_fragment: tower
+'''
+
+RETURN = '''
+tower_version:
+    description: The version of Tower we connected to
+    returned: If connection to Tower works
+    type: string
+job_info:
+    description: dictionnary containing information about the workflow executed
+    returned: If workflow launched
+    type: dict
+'''
+
+
+EXAMPLES = '''
+- name: Launch a workflow
+  tower_workflow_launch:
+    name: "Test Workflow"
+  delegate_to: localhost
+  run_once: true
+  register: workflow_results
+
+- name: Launch a Workflow with parameters without waiting
+  tower_workflow_launch:
+    workflow_template: "Test workflow"
+    extra_vars: "---\nmy: var"
+    wait: False
+  delegate_to: localhost
+  run_once: true
+  register: workflow_task_info
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+
+from ansible.module_utils.ansible_tower import tower_argument_spec, tower_auth_config, HAS_TOWER_CLI
+
+try:
+    import tower_cli
+    from tower_cli.api import client
+    from tower_cli.conf import settings
+    from tower_cli.exceptions import ServerError, ConnectionError, BadRequest, TowerCLIError
+except ImportError:
+    pass
+
+
+def main():
+    argument_spec = tower_argument_spec()
+    argument_spec.update(dict(
+        workflow_template=dict(required=True),
+        extra_vars=dict(required=False),
+        wait=dict(required=False, default=True, type='bool'),
+        timeout=dict(required=False, default=None, type='int'),
+    ))
+
+    module = AnsibleModule(argument_spec=argument_spec, supports_check_mode=True)
+
+    if not HAS_TOWER_CLI:
+        module.fail_json(msg='ansible-tower-cli required for this module')
+
+    workflow_template = module.params.get('workflow_template')
+    extra_vars = module.params.get('extra_vars')
+    wait = module.params.get('wait')
+    timeout = module.params.get('timeout')
+
+    # If we are going to use this result to return we can consider ourselfs changed
+    result = dict(
+        changed=False,
+        msg='initial message'
+    )
+
+    tower_auth = tower_auth_config(module)
+    with settings.runtime_values(**tower_auth):
+        # First we will test the connection. This will be a test for both check and run mode
+        # Note, we are not using the tower_check_mode method here because we want to do more than just a ping test
+        # If we are in check mode we also want to validate that we can find the workflow
```

Comment:
@wwitzel3 isn't this going above and beyond what the check mode performs for other modules?

Reply:
I think some other changes to the config have landed that should certainly eliminate the need for this, in particular the `TowerModule` class.